### PR TITLE
Type hint backwards compitability python version

### PR DESCRIPTION
**Pull request by Sciemon**
This will ensure the compatibility with python version <3.10

The fix_tests branch should be merged first. Then only few changes should occur in this branch.

### CODE REVIEW

File: src/powfacpy/base_interface.py [hunk 1]
```diff
@@ -401,29 +400,23 @@ def delete_obj(self, obj_or_path, condition=None, parent_folder=None, error_if_n
...
-      """
+      o.Delete()
       # 'IsDeleted' seems to be the savest way to check whether an object has been deleted. 
       if not o.IsDeleted():
-        try:
-          o.Deactivate()
-          o.Delete()
-        except:
-          raise ArgumentError(r"Object {o} cannot be deleted.")
+        if not o.IsDeleted():
```
Comment (bertrtam00) [inline src/powfacpy/base_interface.py:406]: merge conflicts with last pull request -- resolve manually, as discussed.

File: src/powfacpy/script_interface.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def merge_file_into_embedded_script(self,
     """
     compython = self.handle_single_pf_object_or_path_input(compython)
     temporary_dir = "powfacpy_temp"
-    os.makedirs(temporary_dir)
+    os.makedirs(temporary_dir,exist_ok=True)
```
Comment (bertrtam00) [inline src/powfacpy/script_interface.py:50]:
```suggestion
    os.makedirs(temporary_dir, exist_ok=True)
```

**Review by bertrtam00 — APPROVED** [review]
thanks! 
apart from the merge conflict I only found a missing space :P (see suggestion)

File: src/powfacpy/base_interface.py (hunk 1 above)
Comment (Sciemon) [reply src/powfacpy/base_interface.py]: Not relevant with latest commit.

**Pull request merged**